staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager_test.go

```diff
@@ -54,104 +53,116 @@ type fakeObjectDefaulter struct{}
 
 func (d *fakeObjectDefaulter) Default(in runtime.Object) {}
 
-func NewTestFieldManager() *fieldmanager.FieldManager {
+type TestFieldManager struct {
+	fieldManager *fieldmanager.FieldManager
+	liveObj      runtime.Object
+}
+
+func NewTestFieldManager() TestFieldManager {
 	gv := schema.GroupVersion{
 		Group:   "apps",
 		Version: "v1",
 	}
 
-	f, _ := fieldmanager.NewCRDFieldManager(
+	f, err := fieldmanager.NewCRDFieldManager(
 		nil,
 		&fakeObjectConvertor{},
 		&fakeObjectDefaulter{},
 		gv,
 		gv,
 		true,
 	)
-	return f
+	if err != nil {
+		panic(err)
+	}
+	return TestFieldManager{
+		fieldManager: f,
+		liveObj:      &unstructured.Unstructured{},
+	}
+}
+
+func (f *TestFieldManager) Reset() {
+	f.liveObj = &unstructured.Unstructured{}
+}
+
+func (f *TestFieldManager) Apply(obj []byte, manager string, force bool) error {
+	var err error
+	f.liveObj, err = f.fieldManager.Apply(f.liveObj, obj, manager, force)
+	return err
+}
+
+func (f *TestFieldManager) Update(obj runtime.Object, manager string) error {
+	var err error
+	f.liveObj, err = f.fieldManager.Update(f.liveObj, obj, manager)
+	return err
+}
+
+func (f *TestFieldManager) CompareObject(obj runtime.Object) error {
```

Comment:
Is this used?

Reply:
Interesting to see that we're never checking that the object is what we would expect. I'll remove it for now, especially since I'm not exactly sure how to implement it yet anyway.